lgc/builder/InOutBuilder.cpp

```diff
@@ -313,7 +313,7 @@ void InOutBuilder::markGenericInputOutputUsage(bool isOutput, unsigned location,
   if (!isOutput || m_shaderStage != ShaderStageGeometry) {
     bool keepAllLocations = false;
     if (getPipelineState()->isUnlinked()) {
-      if (m_shaderStage == ShaderStageVertex && isOutput)
+      if (isOutput && m_pipelineState->getNextShaderStage(m_shaderStage, true) == ShaderStageFragment)
```

Comment:
The original condition makes `keepAllLocations` true for VS output with TCS/GS/FS as the next shader stage. Current condition works for TES and VS output with FS as the next shader stage, is it expected?

Reply:
Yes, that is what I wanted.  The unlinked shaders assume that if any vertex processing stage (VS, TCS, GS, TES) is provided, then no new vertex processing stage will be provided while linking.  If we have both the VS and say GS, then we can change the outputs of the VS to match the inputs of the GS, since we have both of them.  The same for TCS.

Is that correct? Can the VS output be used by the FS when other stages are used?

Reply:
I think FS recieves the output of last vertex processting stage instead of any vertex processing stage, is it right? @amdrexu

Reply:
Yes, only the outputs of last vertex processing stage will be used to match the inputs of fragment shader.

Reply:
Cool, so I think this is fine.  I'll resolve this converstaion.